008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,709 @@
+
+\subsection{Forecasting}
+
+\subsubsection{\textit{Theory}}
+
+Treaties do not have any immediate effect. They instead influence the future of an agent. In order to handle treaties today, agents thus greatly benefit from an ability to predict their future.
+
+In the case of this tower, “future” can mean two decisively different things: The future of an agent on the current level and their future on a series of unpredictable levels after that. E.g. an agent very high up in the tower can expect to receive surplus of food in the coming days, but should not expect the same in future reshuffling periods. The future on this floor is best predicted by the previous experience on this floor, while the future thereafter is best predicted by all previous experience in the tower. Accordingly, we propose each agent separately keeps track of the amount of food it receives each day during the current reassignment period as well as since their first day in the tower. This aligns well with the core assumption in cognitive psychology that there are separate systems for long- and short-term memory.\cite{norris2017short}
+
+\subsubsection{\textit{Implementation}}
+To implement forecasting, we use two separate arrays corresponding to long-term and short-term memory, respectively and storing the amount of food received each day. The short-term memory is reset after every reshuffling. A scaled histogram measuring the number of times the agent received a certain amount of food results in a discrete distribution predicting the expected food in the future.
+
+\begin{figure} [h!]
+    \centering
+    \includegraphics[width=0.9\linewidth]{008_team_6_agent_design/SOMAS_forecasting.pdf}
+    \caption{Forecasting of the estimated food received in the short-term and long-term using two integer arrays.}
+    \label{fig:forecasting}
+\end{figure}
+
+In order to distinguish between the short and long term, this agent also forecasts how long it will stay on the current floor. It does so by averaging the duration of previous reshuffling periods.
+
+\subsection{Utility Theory}
+
+Memory allows this agent to remember the past; Forecasting uses this to predict the future. In order to make a decision regarding treaties, one last step is required. Our agent needs to be able to compare two potential futures and decide which one it deems more beneficial. Behavioural research has shown that the utility a person gets gets from receiving something is not linear to the value of that thing and differs from. We thus use utility functions in our agents to model human decision making more accurately.\cite{fishburn1970utility}
+
+Using utility functions, comparing which one of two different futures is more beneficial is simple. An agent calculates the expected utility with and without a treaty. It then maximise their estimated future benefit by choosing the larger expected utility.
```

Comment:
Suggestion: Using utility functions simplifies the process of comparing the benefit gained from different futures.